File: src/TwitchCommunity.Web/ClientApp/src/app/viewer-games/state/enlistment.model.ts
import { EnlistmentState } from '../models/enlistment-state';

export interface Enlistment {
  id: number | string;
  userName: string;
  state: EnlistmentState;
}

export function createEnlistment(params: Partial<Enlistment>) {
  return {} as Enlistment;
}

File: src/StreamCommunity.Api/ViewerGames/EnlistmentModel.cs
﻿using System;

namespace StreamCommunity.Api.ViewerGames
{
    public class EnlistmentModel
    {
        public int Id { get; set; }

        /// <summary>
        /// Gets or sets the username.
        /// </summary>
        public string UserName { get; set; }

        /// <summary>
        /// Gets or sets the timestamp the enlistment was created.
        /// </summary>
        public DateTime Timestamp { get; set; }

        /// <summary>
        /// Gets or sets the state of the enlistment.
        /// </summary>
        public EnlistmentState State { get; set; }
    }
}
